eth/tracers/api.go

```diff
@@ -1043,3 +1050,54 @@ func overrideConfig(original *params.ChainConfig, override *params.ChainConfig)
 
 	return copy, canon
 }
+
+// argsToTransaction produces a Transaction object given call arguments.
+// The `msg` field must be converted from the same arguments.
+func argsToTransaction(b Backend, args *ethapi.TransactionArgs, msg *core.Message) (*types.Transaction, error) {
```

Comment:
This is very not elegant, since `(*ethapi.TransactionArgs).toTransaction`  already exists, it's just not public:
```
func (args *TransactionArgs) toTransaction() *types.Transaction {
```
It's in `internal/ethapi`. The one over there has support for blob transactions, which this one does not -- which kind of demonstrates how inherently brittle things become when we copy things like this and do not use a single implementation. 

Not sure how to best solve this. @fjl what do you think?

Reply:
Here's how that would look, btw: feel free to cherry-pick if we want to go this way

https://github.com/holiman/go-ethereum/commit/d1e6c8f4fe6ea99102ba91c45416a5a86e7ad908

Reply:
Yeah not sure why I did this. Seems like we even had the public ToTransaction method until [recently](https://github.com/ethereum/go-ethereum/commit/ac5aa672d3b85a1f74667a65a15398f072aa0b2a#diff-4a9c1433c7c79c50cb72fd743c896f7bf5a976f46ccbb5fae83f5c46b2465db9L406-L410). Going to re-add that and use.

Reply:
Aha I see why I did this: because just prior to this we call `args.ToMessage` and already do all the conversions from hex types. I wanted to save that overhead on `args.ToTransaction` and re-use most fields from the converted `Message`. I'm now thinking it's not worth the performance gain. So I'll use `args.ToTransaction` directly.

Reply:
I don't get it. 

> I re-used the message field ...

Where? Is that some commit that you haven't pushed yet?

Reply:
I've now pushed a commit addressing this. Allow me to reformulate my rambling from above:

TraceCall receives the TransactionArgs from user. Previously the args had to be converted to a Message so EVM can process it. Now because we also pass the `*types.Transaction` object within EVM, we have to convert the args both to a message and a transaction. And obviously we don't want them to be inconsistent.

The args to message conversion previously also filled in some fields with 0 values (with the exception of gaslimit which it set to global gas cap). I've now moved that logic to a new method on TransactionArgs (calling it `CallDefaults` because it is used for eth_call, debug_traceCall etc.) that fills in these defaults. This allows me to call `ToMessage` and `ToTransaction` on it safely. Note it's not ideal to use `setDefaults` for this because it needs the `ethapi.Backend` which is much larger than the tracing backend.